### PR TITLE
Allows scalac ignores java errors.

### PR DESCRIPTION
**Pull request by wpopielarski**
Java errors are reported to `SbtBuildReporter` by java compiler
directly.
Commit removes a bug when java depends on scala code (see `SbtInputs`)

Fixes #1002738

### CODE REVIEW

File: org.scala-ide.sdt.core/src/org/scalaide/core/internal/builder/zinc/SbtBuildReporter.scala [hunk 1]
```diff
@@ -88,7 +109,7 @@ private[zinc] class SbtBuildReporter(project: IScalaProject) extends xsbti.Repor
       val markerPos = MarkerFactory.RegionPosition(offset, identifierLength(pos.lineContent, pos.pointer), line)
       BuildProblemMarker.create(resource, severity, msg, markerPos)
     } else
-      logger.error(s"suppressed error in Java file ${resource.getFullPath}:$line: $msg")
+      logger.warn(s"suppressed error in Java file ${resource.getFullPath}:$line: $msg")
```
Comment (sschaef) [inline org.scala-ide.sdt.core/src/org/scalaide/core/internal/builder/zinc/SbtBuildReporter.scala]: shouldn't it just log to `info` now? Afterwards javac reports the errors now as far as I understand.

File: org.scala-ide.sdt.core/src/org/scalaide/core/internal/builder/zinc/SbtInputs.scala [hunk 2]
```diff
@@ -47,13 +47,15 @@ class SbtInputs(installation: IScalaInstallation,
...
+      Maybe.nothing[Analysis]
     else {
       val analysis = allProjects.collectFirst {
         case project if project.buildManager.buildManagerOf(f).nonEmpty =>
           project.buildManager.buildManagerOf(f).get.latestAnalysis(incOptions)
       }
-      Maybe.just(analysis.getOrElse(Analysis.Empty))
+      analysis.map { analysis =>
+        Maybe.just(analysis)
+      }.orElse(Option(Maybe.nothing[Analysis])).get
```
Comment (sschaef) [inline org.scala-ide.sdt.core/src/org/scalaide/core/internal/builder/zinc/SbtInputs.scala]: `getOrElse` instead of `orElse.get`?

Comment (wpopielarski) [inline org.scala-ide.sdt.core/src/org/scalaide/core/internal/builder/zinc/SbtInputs.scala]: agree but it was intentional. I thought rather about operating on same type. Doesn't matter, simplify it

**Comment by sschaef** [comment]
LGTM

**Pull request merged**